internal/controller/helper_test.go

```diff
@@ -47,31 +49,18 @@ func testBuildInMemoryCredentialsSecret(name, ns string) *v1.Secret {
 	}
 }
 
-func getTestEndpoints(dnsName string) []*externaldnsendpoint.Endpoint {
-	return []*externaldnsendpoint.Endpoint{
-		{
-			DNSName: dnsName,
-			Targets: []string{
-				"127.0.0.1",
-			},
-			RecordType:       "A",
-			SetIdentifier:    "",
-			RecordTTL:        60,
-			Labels:           nil,
-			ProviderSpecific: nil,
-		},
-	}
+func getDefaultTestEndpoints() []*externaldnsendpoint.Endpoint {
+	return getTestEndpoints("foo.example.com", "127.0.0.1")
 }
-
-func getTestNonExistingEndpoints() []*externaldnsendpoint.Endpoint {
+func getTestEndpoints(dnsName, ip string) []*externaldnsendpoint.Endpoint {
 	return []*externaldnsendpoint.Endpoint{
 		{
-			DNSName: "nope.example.com",
+			DNSName: dnsName,
 			Targets: []string{
-				"127.0.0.1",
+				ip,
 			},
 			RecordType:       "A",
-			SetIdentifier:    "",
+			SetIdentifier:    "foo",
```

Comment:
Why add the SetIdentifier?

Personally i think these "helper" functions just lead to confusion in the tests. There is no real one size fits all endpoint list and we should probably be explicitly defining what we expect it to be for each spec.

Reply:
I think it'd be best to convert this to a fixture, like this: https://github.com/Kuadrant/dns-operator/blob/main/test/e2e/healthcheck_test.go#L62 and this: https://github.com/Kuadrant/dns-operator/blob/main/test/e2e/fixtures/healthcheck_test/geo-dnsrecord-healthchecks.yaml

Reply:
> I think it'd be best to convert this to a fixture, like this: https://github.com/Kuadrant/dns-operator/blob/main/test/e2e/healthcheck_test.go#L62 and this: https://github.com/Kuadrant/dns-operator/blob/main/test/e2e/fixtures/healthcheck_test/geo-dnsrecord-healthchecks.yaml

These are the integration tests where a much larger amount of variations in what we are testing would be expected. I'm not sure using fixtures is appropriate here, or would help with the readability of tests.

Reply:
If `SetIdentifier` is an empty string it does not conflict records ([here](https://github.com/Kuadrant/dns-operator/blob/GH-84/internal/external-dns/plan/plan.go#L434-L437)). There is probably a smarter way of doing it

Reply:
The way I read these tests is "we want to use valid/broken endpoints". As a person who isn't yet used to stare into enpoints it made it easier for me to see a func call with parameters that will differ compared to the whole struct each time. 
On the other hand if "helper" functions aren't helping the majority we should definitely remove those.

Reply:
> If `SetIdentifier` is an empty string it does not conflict records ([here](https://github.com/Kuadrant/dns-operator/blob/GH-84/internal/external-dns/plan/plan.go#L434-L437)). There is probably a smarter way of doing it

It won't conflict because it's not supposed to, it will merge the targets, which i kind of my point here. There are lots of permutations of how the endpoints could be configured that trying to write a single function that works for all tests is probably impossible and will just make tests harder to understand.

We can leave this for now, but we should look at doing it differently going forward since the endpoint and endpoint list on the record is the main thing we will be changing in tests to force different scenarios and you want to be able to see what they look like easier than this.